.github/workflows/flathub-pr-check.yml

```diff
@@ -0,0 +1,96 @@
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+concurrency:
+  # auto-cancel previous runs if the pull request is force-push updated
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+jobs:
+  build-flatpak:
+    runs-on: ubuntu-latest
+    steps:
+    # PR/CI
+    - name: Checkout relevant repo
+      uses: actions/checkout@v4
+      with:
+        path: "xbmc"
+    # Flathub config
+    - name: Checkout Flathub config
+      uses: actions/checkout@v4
+      with:
+        repository: 'flole998/tv.kodi.Kodi'
+        submodules: true
+        ref: "beta"
+        path: "flathub"
+    - name: Install dependencies
+      run: |
+        sudo apt-get update -y
+        DEBIAN_FRONTEND=noninteractive sudo apt-get install --force-yes -y cmake git build-essential ccache dbus-x11 flatpak
+    - name: Grant tar root access
+      run: sudo chown root /bin/tar && sudo chmod u+s /bin/tar
+    - id: restore-cache
+      name: Restore cache
+      uses: actions/cache/restore@v4
+      with:
+        path: |
+          /tmp/ccache
+          flathub/.flatpak-builder
+          /var/lib/flatpak
+        key: flatpak-cache-${{ github.run_id }}
+        restore-keys: |
+          flatpak-cache
+    - name: Remove tar root access
+      run: sudo chmod u-s /bin/tar
+    - name: Adjust cache permissions if cache was restored
+      run: |
+        sudo chmod -R 777 /tmp/ccache || true
+        sudo chmod -R 777 flathub/.flatpak-builder || true
+    - name: Start DBUS
+      run: sudo service dbus start
+    - name: Install yq
+      run: sudo wget https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64 -O /usr/bin/yq && sudo chmod +x /usr/bin/yq
+    - name: Add flatpak repo
+      run: sudo flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
+    - name: Install flatpak SDKs
+      run: |
+        sudo flatpak -y install flathub org.freedesktop.Sdk//22.08
+        sudo flatpak -y install flathub org.freedesktop.Platform//22.08
+        sudo flatpak -y install flathub org.freedesktop.Sdk.Extension.openjdk17//22.08
+        sudo flatpak -y install flathub org.flatpak.Builder
+    - name: Patch flatpak build conf
+      run: |
+        # Remove git repo from local copy
+        yq -i eval 'del(.modules[].sources[] | select(.type == "git" and .url == "https://github.com/xbmc/xbmc.git"))' flathub/tv.kodi.Kodi.yml
+        # Add local git repo copy
+        yq -i eval '(.modules[] | select(.name == "kodi") | .sources) |= [{"type": "dir", "path": "'"$GITHUB_WORKSPACE"'/xbmc"}] + .' flathub/tv.kodi.Kodi.yml
+        # Remove addons from build
+        yq -i 'del(.modules[] | select(. == "addons/*")) | .' flathub/tv.kodi.Kodi.yml
+    - name: build
+      env:
+        CCACHE_DIR: /tmp/ccache
+        FLATPAKBUILDER: flatpak run org.flatpak.Builder
+      run: cd flathub && dbus-launch make build
+    - name: flatpak
```

Comment:
I guess this is the flatpak build?

Reply:
This creates the .flatpak file, yes.

Reply:
I guess I'm asking for making the step names a bit more explicit then

Reply:
Sure, do you know if there's a "proper" name for these operations? The first one is "Build Kodi" I assume, this one is "Create flatpak-file"?

Reply:
Done

Reply:
I'm still strugging to understand how `dbus-launch make build` causes a kodi build - doing all this manually and not just using the provided github action is sure to make stuff harder to understand. I know @fuzzard liked it being pure shell commands - but I'm having a hard time as that also means we're not on the happy/supported path.

I'm fine with going ahead with this, I just want to mention it. Not sure if there was anything technical actually blocking us from using the provided actions.

Reply:
`make build` is doing the build as specified in the Makefile (using flatpak-builder). However, flatpak relies on dbus, so it needs to be setup/running properly. On a "normal" system you could simply run `make build`, but when doing it in a GitHub actions container this becomes necessary.

I'll have a look at how others are doing it, maybe there's something to learn from it.

The provided actions would now allow kicking out the addons before the build I think?

Reply:
I think you meant to write `not` not `now`.

If that's the case, no, I think we would need to still do most of them (probably minus the dbus stuff and I guess it should also do caching for us)

Reply:
Yes, "not" not "now".

It does caching, yes. My manual approach is still 8 minutes faster for some reason, I assume it is due to caching.

I don't think you can run commands between checkout and doing the build when using the "magic" action. It does everything under the hood, which is fine until you want to customize stuff (like removing addons before building).

Reply:
I might be missing something then, I would expect it to just work fine, if the docs here are correct https://github.com/flathub-infra/flatpak-github-actions/blob/master/README.md?plain=1#L26

Why wouldn't you be able to change what you checked out in the line before?

Reply:
And if we can figure out how to improve the caching of the action too, that would be huge for everyone involved with flatpak :)